### PR TITLE
Enable linker warnings as errors

### PR DESCRIPTION
**Pull request by devpow112**

### CODE REVIEW

File: Overlays/Windows/cmake/build_options.cmake [hunk 1]
```diff
@@ -76,6 +76,8 @@ add_compile_options(/D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
 add_compile_options(/D_WN_WINDOWS)
 
 # Linker options
+add_link_options(/WX) # Enable warnings as errors
```
Comment (devpow112) [inline Overlays/Windows/cmake/build_options.cmake:79]: Doesn't work for static libs with MSVC. Will change.

**Comment by devpow112** [comment]
THis can be enabled after #220 is merged.

**Review by AWoloszyn — APPROVED** [review]

**Comment by devpow112** [comment]
Re-run after https://github.com/WNProject/DockerBuildWindows/pull/29 is merged/deployed.

**Review by AWoloszyn — DISMISSED** [review]

**Review by AWoloszyn — APPROVED** [review]

**Review by AWoloszyn — APPROVED** [review]

**Pull request merged**